src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -366,6 +365,80 @@ private void checkIfGroupsExists(Course course) {
         return ResponseUtil.wrapOrNotFound(Optional.ofNullable(course));
     }
 
+    /**
+     * GET  /courses/:id/with-exercises-and-relevant-participations
+     *
+     * Get the "id" course, with text and modelling exercises and their participations
+     *
+     * @param id the id of the course to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the course, or with status 404 (Not Found)
+     */
+    @GetMapping("/courses/{id}/with-exercises-and-relevant-participations")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Course> getCourseWithExercisesAndRelevantParticipations(@PathVariable Long id) {
+        log.debug("REST request to get Course with exercises and relevant participations : {}", id);
+        Course course = courseService.findOneWithExercises(id);
+
+        if (!userHasPermission(course)) return forbidden();
```

Comment:
please use the new Exception model for this case

Reply:
Done